packages/frontend/src/pages/NewListing.tsx

```diff
@@ -0,0 +1,421 @@
+import { z } from 'zod'
+
+//TODO: ✅ Add field validation
+//TODO: Hook up to Trustlist hook (maybe?) (if it needs to exist)
+//TODO: User must be logged in to add listing
+//TODO: When do we calculate trust scores and how? Seems like before sending the formData we calc it but ask to be sure
+// TODO: Choosing what epoch key — I don't think this needs to be a user selected thing. Whats the difference between them choosing one long string vs another? Basically a coin flip right?
+
+const trustScoresFromData = { ...trustScores }; // Make copy we can use
+
+const NewListingResponseSchema = z.object({
+  epoch: z.string(),
+  categories: z.record(z.string().min(1, 'Please choose an option')),
+  title: z.string().min(1, 'Please add a title'),
+  price: z.number().min(1, 'Please add the price'), //TODO: usd now , include crypto (?)
+  frequency: z.literal("one time"),
+  description: z.string().min(1, 'Please describe what you are listing'),
+  posterId: z.string(),
+  revealTrustScores: z.record(z.boolean()),
+  // scores: z.record(z.number())
+})
+
+export type NewListingResponse = z.infer<typeof NewListingResponseSchema>
+
+type FormStep = {
+  id: string,
+  description?: string,
+  fields: string[]
+  validate?: (data: NewListingResponse) => boolean,
+}
+
+const FormSteps: FormStep[] = [
+  { id: 'select-category', description: 'Choose the categories for your listing', fields: ['categories'] },
+  { id: 'general-info', description: 'Enter general information', fields: ['title', 'description', 'price'] },
+  { id: 'trust-scores', description: 'Choose what trust scores to show', fields: ['revealTrustScores'] },
+]
+
+type FormState = {
+  fields: NewListingResponse
+  step: FormStep
+  isLoading: boolean
+  error?: string
+}
+
+type ListingFormValues = FieldValues & NewListingResponse
+
+type StepSectionProps = UseFormReturn<ListingFormValues>
+
+type FormFooterAndHeaderProps = { currentStep: number, changeStep: React.Dispatch<React.SetStateAction<FormStep>>, trigger: UseFormTrigger<ListingFormValues> }
+
+
+const createInitialCategories = (sectionsWithCategories: Record<string, string[]>) => {
+  // create an empty array for each section in listing type
+  return Object.keys(sectionsWithCategories).reduce<Record<string, string>>((acc, key) => {
+    acc[key] = '';
+    return acc;
+  }, {});
+}
+
+const initialCategories = createInitialCategories(listingCategories);
+
+const initialFormState: FormState = {
+  fields: {
+    epoch: '',
+    categories: initialCategories,
+    title: '',
+    price: 0,
+    frequency: 'one time',
+    description: '',
+    posterId: '',
+    revealTrustScores: {
+      [TrustScoreKeyEnum.LP]: true,
+      [TrustScoreKeyEnum.LO]: true,
+      [TrustScoreKeyEnum.CB]: true,
+      [TrustScoreKeyEnum.GV]: true,
+    },
+    // scores: {
+    //   [TrustScoreKeyEnum.LP]: 0.3,
+    //   [TrustScoreKeyEnum.LO]: 0.2,
+    //   [TrustScoreKeyEnum.CB]: 0.03,
+    //   [TrustScoreKeyEnum.GV]: 0.5,
+    // }
```

Comment:
i think we can remove `revealTrustScores` from the listing (and `isRevealed` from the record in `data`)
.. bc we shouldn't store scores that the user wants to hide. the idea is for the listing to show a (chosen to be) revealed score, or that the score is not being revealed, or that the user has not accumulated any data for that metric. so i'd like to just add the score as a string, it can be either the score, or an X if hidden, or n/a if no data.
.. next comment 👇